src/huggingface_hub/repocard.py

```diff
@@ -725,7 +727,19 @@ def metadata_update(
         else "Update metadata with huggingface_hub"
     )
 
-    card = ModelCard.load(repo_id, token=token)
+    # Card class given repo_type
+    card_class: Type[RepoCard] = RepoCard
+    if repo_type is None or repo_type == "model":
```

Comment:
So in this case we use Repocard under the hood if they say something other than model or dataset? Should we warn in that case as well, or nah?

Reply:
That would be the case for `space` repo_type. I am not sure what warning message to put but I'm fine with using `ModelCard` by default as before. I don't think it changes anything as the card is created and then uploaded directly. The python object itself it never really used. As you prefer :)

Reply:
If creating the card for space repo, a blank one would probably break things on spaces, eh? Since it has no way of knowing the sdk, app file, etc.

Reply:
:exploding_head:

Reply:
I think a readme is created by default, eh? So one would have to delete manually in order to screw things up. Lemme run a test real quick in colab and comment here in a few

Reply:
Ah, right now this code will actually break if you call with `repo_type='space'` and the repo exists. 

For example - this repo.

```python
from huggingface_hub import metadata_update

metadata_update('nateraw/deleteme-test-space-card-update', {'tags': 'asdf'}, repo_type='space')
```

Reply:
> Ah, right now this code will actually break if you call with repo_type='space' and the repo exists.
Oopsi, it turns out `repo_type` was not used to load the card

Discovered another issue at the same time. Emoji been unicode-ified:
https://huggingface.co/spaces/nateraw/deleteme-test-space-card-update/discussions/1/files

Reply:
@nateraw I updated the code.

Now:
- if readme doesn't exist and repo type is "model" or "None" -> create one from ModelCard template
- if readme doesn't exist and repo type is "dataset" -> create one from DatasetCard template
- if readme doesn't exist and repo type is "space" -> raise exception -> should actually never happen as the file is created automatically.

I added a test for each  `repo_type`. Should be fine now ? :) 

(also to mention, I fixed the unicode dump in this PR as well. I felt a bit lazy to create a separate one)